src/sql/workbench/services/notebook/browser/models/modelInterfaces.ts

```diff
@@ -461,6 +461,8 @@ export interface ICellModel {
 	readonly onCellModeChanged: Event<boolean>;
 	modelContentChangedEvent: IModelContentChangedEvent;
 	isEditMode: boolean;
+	showPreview?: boolean;
```

Comment:
So ICellModel is only implemented in one place - the CellModel class you have above. So making this non-nullable isn't going to affect any existing code and is what I would suggest doing.